### PR TITLE
feat(cypress): add support for cypress-multi-reporter

### LINKED ISSUE #701: [cypress] add support for cypress-multi-reporter
We're using the https://www.npmjs.com/package/cypress-multi-reporters Cypress plugin, which takes in a config file that lists other reporters.

Example,
```ts 
// in cypress.config.ts
export default defineConfig({
  reporter: 'cypress-multi-reporters',
  reporterOptions: {
    configFile: 'reporter-config.json',
  },
  // other stuff
})
```

and in the referenced config file,
```json
{
  "reporterEnabled": "mochawesome, mocha-junit-reporter", // comma + space separated list
}
```

Which currently reports,
```
## Unused dependencies (3)

| Name                    | Location     | Severity |
| :---------------------- | :----------- | :------- |
| cypress-multi-reporters | package.json | error    |
| mocha-junit-reporter    | package.json | error    |
| mochawesome             | package.json | error    |
```

With the following package.json dependencies,
```json
{
"name": "my-e2e-package",
"dependencies": {
    "cypress": "12.17.4",
    "cypress-multi-reporters": "^1.6.3",
    "mocha": "^10.2.0",
    "mocha-junit-reporter": "^2.2.0",
    "mochawesome": "^6.2.2"
  }
}
```

It would be awesome to see this supported in knip! Happy to submit a PR!

### PR DESCRIPTION
**Pull request by aramissennyeydd**
Closes #701

### CODE REVIEW

File: packages/knip/src/plugins/cypress/types.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface CypressConfig {
```
Comment (aramissennyeydd) [inline packages/knip/src/plugins/cypress/types.ts:1]: Should I add the cypress types here instead? Looks like that would pull in the entire cypress dependency

File: packages/knip/test/plugins/cypress-multi-reporter.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+test('Find dependencies with the cypress-multi-reporter plugin', async () => {
+  const { issues, counters } = await main({
+    ...baseArguments,
+    cwd,
+  });
+
+  assert(issues.unlisted['cypress.config.ts']['@nrwl/cypress/plugins/cypress-preset']);
+  assert(issues.unlisted['cypress/support/commands.ts']['@faker-js/faker']);
+  assert(issues.unlisted['cypress/support/e2e.ts']['@testing-library/cypress/add-commands']);
+  assert(issues.unlisted['cypress.config.ts']['@testing-library/my-fake-reporter']);
```
Comment (aramissennyeydd) [inline packages/knip/test/plugins/cypress-multi-reporter.test.ts:19]: Is there anyway I can make this report as 
```ts
assert(issues.unlisted['reporter-config.json']['@testing-library/my-fake-reporter']);
```
instead?

File: packages/knip/src/plugins/cypress/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  // https://github.com/YOU54F/cypress-plugins/tree/master/cypress-multi-reporters#configuring-reporters
+  if (reporter === 'cypress-multi-reporters' && config.reporterOptions?.configFile) {
+    // Try to resolve the config file if present and attach the reporters listed in it.
+    const { configFile } = config.reporterOptions;
+    const configFilePath = toAbsolute(configFile, configFileDir);
+    if (isInternal(configFilePath)) {
+      const reporterConfig = await load(configFilePath);
+      if (typeof reporterConfig === 'object' && reporterConfig.reporterEnabled) {
+        const { reporterEnabled: reporterConcatenatedNames } = reporterConfig;
+        const reporterNames = reporterConcatenatedNames.split(', ');
```
Comment (webpro) [inline packages/knip/src/plugins/cypress/helpers.ts:23]: Maybe allow without spaces too? eg:

```
.split(/, ?/)
```

File: packages/knip/src/plugins/cypress/types.ts (hunk 1 above)
Comment (webpro) [reply packages/knip/src/plugins/cypress/types.ts:1]: Hoping this isn't necessary. Typically I've held off and tried to "inline" only the required types. Do you think that's feasible here too?

File: packages/knip/test/plugins/cypress-multi-reporter.test.ts (hunk 2 above)
Comment (webpro) [reply packages/knip/test/plugins/cypress-multi-reporter.test.ts:19]: Unfortunately this isn't supported (yet).

**Review by webpro — COMMENTED** [review]
Thanks for the pull request! Looks great, added a few comments. No biggies, just hoping we don't have to pull in the entire cypress dependency.

File: packages/knip/src/plugins/cypress/helpers.ts (hunk 3 above)
Comment (aramissennyeydd) [reply packages/knip/src/plugins/cypress/helpers.ts:23]: Good call, looks like they use `.trim()` in their code (https://github.com/YOU54F/cypress-plugins/blob/master/cypress-multi-reporters/lib/MultiReporters.js#L50-L58). Let me try that here as well.

File: packages/knip/src/plugins/cypress/types.ts (hunk 1 above)
Comment (aramissennyeydd) [reply packages/knip/src/plugins/cypress/types.ts:1]: Yup, let's stick with the current inlined types!

File: packages/knip/src/plugins/cypress/helpers.ts (hunk 3 above)
Comment (aramissennyeydd) [reply packages/knip/src/plugins/cypress/helpers.ts]: Updated in [e038b50](https://github.com/webpro-nl/knip/pull/726/commits/e038b50a37eaaf610f03489b9bde1bdf379e03d1)

**Comment by aramissennyeydd** [comment]
@webpro I don't _think_ the new test errors are related to this PR, any ideas what's happening here?

**Comment by webpro** [comment]
> @webpro I don't _think_ the new test errors are related to this PR, any ideas what's happening here?

The tests are stable in my experience, so they should just pass.

I do see some imports having the `.ts` extension instead of `.js`.

**Comment by aramissennyeydd** [comment]
@webpro Yup that was it, looks like everything passed once I changed those!

File: packages/knip/src/plugins/cypress/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        // Pulled from the reporter source code, https://github.com/YOU54F/cypress-plugins/blob/master/cypress-multi-reporters/lib/MultiReporters.js#L50-L58
+        // Not sure why they allow for extra whitespace characters, but let's handle it the same as them.
+        const reporterNames = reporterConcatenatedNames.split(',');
+        for (const reporterName of reporterNames) {
+          reporters.push(resolve(reporterName.trim()));
+        }
+      }
+    }
+  }
+  return [...reporters];
```
Comment (webpro) [inline packages/knip/src/plugins/cypress/helpers.ts:36]: Why the spread here? Maybe leave it but make `reporters` a Set to prevent duplicates?

**Review by webpro — APPROVED** [review]
Minor thingie, happy to merge either way!

File: packages/knip/src/plugins/cypress/helpers.ts (hunk 4 above)
Comment (aramissennyeydd) [reply packages/knip/src/plugins/cypress/helpers.ts:36]: Was using the jest plugin as an example. Updated this to be a Set!

File: packages/knip/src/plugins/cypress/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+export const resolveDependencies = async (config: CypressConfig, options: PluginOptions) => {
+  const { reporter } = config;
+  const { configFileDir } = options;
+
+  const resolve = (specifier: string) => resolveEntry(options, specifier);
+
+  // Initialize the array of reporters with the initial reporter if present.
+  const reporters = reporter ? new Set(reporter) : new Set();
```
Comment (webpro) [inline packages/knip/src/plugins/cypress/helpers.ts:17]:
```suggestion
  const reporters = reporter ? new Set([reporter]) : new Set();
```

**Comment by aramissennyeydd** [comment]
🌬️ away with ye type error

File: packages/knip/src/plugins/cypress/helpers.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+export const resolveDependencies = async (config: CypressConfig, options: PluginOptions) => {
+  const { reporter } = config;
+  const { configFileDir } = options;
+
+  const resolve = (specifier: string) => resolveEntry(options, specifier);
+
+  // Initialize the array of reporters with the initial reporter if present.
+  const reporters: Set<string> = reporter ? new Set(reporter) : new Set();
```
Comment (webpro) [inline packages/knip/src/plugins/cypress/helpers.ts:17]:
```suggestion
  const reporters: Set<string> = reporter ? new Set([reporter]) : new Set();
```

**Comment by webpro** [comment]
Thanks for the efforts, @aramissennyeydd! Appreciated.

**Pull request merged**